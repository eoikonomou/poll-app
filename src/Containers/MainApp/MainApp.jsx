import React from 'react';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';

import PollOptions from '../PollOptions';
import PollVote from '../PollVote';
import PollStats from '../PollStats';
import './MainApp.scss';

const MainApp = observer(({ store }) => (
  <>
    <PollOptions store={store} />
    <PollVote store={store} />
    <PollStats store={store} />
  </>
));

MainApp.displayName = 'MainApp';

MainApp.propTypes = {
  store: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.array,
    PropTypes.bool,
  ])).isRequired,
};

export default MainApp;
