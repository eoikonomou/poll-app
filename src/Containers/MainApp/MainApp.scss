.section {
  padding: 10px;
  flex: 1;

  .section-title {
    font-size: 24px;
    font-weight: 900;
    padding-bottom: 10px;
    color: #EED179;
  }
}

.flex-container {
  display: flex;
  flex-direction: column;
  padding-top: 5px;
}
