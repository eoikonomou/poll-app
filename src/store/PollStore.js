import { action, observable } from 'mobx';

export default class PollStore {
  @observable pollQuestion = '';

  @observable pollOptions = [];

  @observable pollLocked = false;

  @action setPollQuestion = (data) => {
    this.pollQuestion = data;
  }

  @action setPollOptions = (data) => {
    this.pollOptions = data;
  }

  @action addPollOption = (data) => {
    this.pollOptions.push(data);
  }

  @action addVote = (optionId) => {
    const option = this.pollOptions.find((opt) => opt.id === optionId);
    option.count += 1;
  }

  @action reset = () => {
    this.pollOptions = [];
    this.pollQuestion = '';
    this.pollLocked = false;
  }

  @action setPollLocked = (data) => {
    this.pollLocked = data;
  }
}
