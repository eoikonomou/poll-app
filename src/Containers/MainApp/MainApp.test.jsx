import React from 'react';
import { shallow } from 'enzyme';
import MainApp from './MainApp';
import PollStore from '../../store/PollStore';

describe('MainApp', () => {
  test('should render successfully', () => {
    const store = new PollStore();
    const wrapper = shallow(<MainApp store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
