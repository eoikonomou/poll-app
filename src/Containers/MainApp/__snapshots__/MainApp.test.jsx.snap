// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp should render successfully 1`] = `
<Fragment>
  <PollOptions
    store={PollStore {}}
  />
  <PollVote
    store={PollStore {}}
  />
  <PollStats
    store={PollStore {}}
  />
</Fragment>
`;
